### PR TITLE
264 - Added functional and e2e tests for Tree

### LINKED ISSUE #264: SOHO-8165: Tree Functional and E2E Tests
### Original ticket: [SOHO-8165](https://jira.infor.com/browse/SOHO-8165)

Add some tests for Tree

See existing tests for example textarea, dropdown, treemap, menubutton for reference. Also see docs/TESTING.MD for commands. Ask for help on sohoxi-dev if needed.

Functional Tests
- Test events that get triggered
- Test events that are listened to
- Test all API methods, most importantly the public ones first and the rest for coverage. (fx enable/disable)
- Test lifecycle methods (init, teardown, destroy, invoke)
- Test setting and that some states are applied

E2e Tests
- Load each test page in the components/<name>/list and test the main ideas of the page, if unsure of the main idea ask on the channel.
- Keyboard

Tree Grid Specific Test Considerations 
see all of http://localhost:4000/components/tree/list examples and tests

Other Tests
- Accessibility tests https://github.com/dequelabs/axe-core
- Security Tests (TBD)
- Specific Test Cases from Cindy (TBD)
- Protractor Image Comparison Tests  (TBD)

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Added functional and e2e tests for Tree

**Related github/jira issue (required)**:
Closes #264 (formerly) https://jira.infor.com/browse/SOHO-8165

**Steps necessary to review your pull request (required)**:
E2E
`env PROTRACTOR_SPECS=‘components/tree/tree.e2e-spec.js' npm run e2e:local:debug`
`env npm run e2e:local:debug`
FUNCTIONAL
`npm run functional:local`

### CODE REVIEW

File: test/components/tree/tree.e2e-spec.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    expect(await element.all(by.css('.tree li.folder a[role="treeitem"]')).get(0).element(by.css('.icon-tree use')).getAttribute('xlink:href')).toContain('#icon-closed-folder');
+    await browser.driver.sleep(config.sleep);
+    await element.all(by.css('.tree li.folder a[role="treeitem"]')).get(0).click();
+
+    expect(await element.all(by.css('.tree li.folder')).get(0).getAttribute('class')).toContain('is-open');
+    expect(await element.all(by.css('.tree li.folder a[role="treeitem"]')).get(0).element(by.css('.icon-tree use')).getAttribute('xlink:href')).toContain('#icon-open-folder');
+  });
+});
+
+describe('Tree bdges tests', () => {
```
Comment (tmcconechy) [inline test/components/tree/tree.e2e-spec.js]: badges

File: test/components/tree/tree.func-spec.js [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+    const secondDir = treeEl.querySelectorAll('li.folder')[2];
+
+    expect(secondDir.querySelectorAll('ul[role=group].is-open').length).toEqual(0);
+
+    treeObj.expandAll($(secondDir.querySelector('ul[role=group]')));
+
+    expect(secondDir.querySelectorAll('ul[role=group].is-open').length).toEqual(1);
+  });
+
+  it('Should expands all tree nodes', () => {
```
Comment (tmcconechy) [inline test/components/tree/tree.func-spec.js]: expand

File: test/components/tree/tree.func-spec.js [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+
+  it('Should collapse all tree nodes', () => {
+    expect(treeEl.querySelectorAll('li.folder ul[role=group].is-open').length).toEqual(2);
+
+    treeObj.collapseAll();
+
+    expect(treeEl.querySelectorAll('li.folder ul[role=group].is-open').length).toEqual(0);
+  });
+
+  it('Should selects a node specifically using its ID attribute', () => {
```
Comment (tmcconechy) [inline test/components/tree/tree.func-spec.js]: select

File: test/components/tree/tree.func-spec.js [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+
+    expect(link.classList.contains('is-selected')).toBeFalsy();
+
+    treeObj.selectNodeById('home');
+    link = treeEl.querySelector('a[role="treeitem"]');
+
+    expect(link.classList.contains('is-selected')).toBeTruthy();
+  });
+
+  it('Should selects a tree node by jquery-selector', () => {
```
Comment (tmcconechy) [inline test/components/tree/tree.func-spec.js]: select

File: test/components/tree/tree.func-spec.js [hunk 5]
```diff
@@ -0,0 +1,588 @@
...
+
+    expect(link.classList.contains('is-selected')).toBeFalsy();
+
+    treeObj.selectNode($(link));
+    link = treeEl.querySelector('a[role="treeitem"]');
+
+    expect(link.classList.contains('is-selected')).toBeTruthy();
+  });
+
+  it('Should deselects a given node', () => {
```
Comment (tmcconechy) [inline test/components/tree/tree.func-spec.js]: deselect

**Comment by deep7102** [comment]
Updated, all the typos now

**Pull request merged**